tests/integration/study_data_blueprint/test_binding_constraints.py

```diff
@@ -1031,3 +1058,153 @@ def test_for_version_870(self, client: TestClient, user_access_token: str, study
+                    assert (df == 0).all().all()
+
+                superfluous_matrices = {"lt", "gt", "eq"} - required_matrices
+                for matrix in superfluous_matrices:
+                    try:
+                        preparer.download_matrix(variant_id, f"input/bindingconstraints/{bc_id}_{matrix}")
+                    except HTTPError as e:
+                        assert e.response.status_code == 404
+                    else:
+                        assert False, "The matrix should not exist"
+
+        # drop all commands to avoid conflicts with the next test
+        preparer.drop_all_commands(variant_id)
+
+        # =============================
+        #  CREATION WITH MATRICES
+        # =============================
+
+        # random matrices
+        matrices_by_time_steps = {
+            time_step: np.random.rand(size, 1).astype(np.float64) for time_step, size in MATRIX_SIZES.items()
+        }
+
+        for bc_time_step in all_time_steps:
+            for bc_operator in all_operators:
+                bc_name = f"BC_{bc_time_step}_{bc_operator}"
+                matrix = matrices_by_time_steps[bc_time_step].tolist()
+                args = {
+                    "name": bc_name,
+                    "type": bc_time_step,
+                    "operator": bc_operator,
+                    "coeffs": {f"{area_id}.{cluster_id.lower()}": [1, 2]},
+                }
+                if bc_operator == "less":
+                    args["lessTermMatrix"] = matrix
+                elif bc_operator == "greater":
+                    args["greaterTermMatrix"] = matrix
+                elif bc_operator == "equal":
+                    args["equalTermMatrix"] = matrix
+                else:
+                    args["lessTermMatrix"] = args["greaterTermMatrix"] = matrix
+                res = client.post(
+                    f"/v1/studies/{variant_id}/commands",
+                    json=[{"action": "create_binding_constraint", "args": args}],
+                )
+                assert res.status_code == 200, res.json()
+
+        preparer.generate_snapshot(variant_id, denormalize=denormalize)
+
+        # Check the matrices size, values and existence
+        for bc_time_step in all_time_steps:
+            for bc_operator in all_operators:
+                bc_name = f"BC_{bc_time_step}_{bc_operator}"
+                bc_id = bc_name.lower()
+
+                required_matrices = REQUIRED_MATRICES[bc_operator]
+                for matrix in required_matrices:
+                    df = preparer.download_matrix(variant_id, f"input/bindingconstraints/{bc_id}_{matrix}")
+                    assert df.shape == (MATRIX_SIZES[bc_time_step], 1)
+                    assert np.allclose(df.values, matrices_by_time_steps[bc_time_step], atol=1e-6)
+
+                superfluous_matrices = {"lt", "gt", "eq"} - required_matrices
+                for matrix in superfluous_matrices:
```

Comment:
don't really understand this check. `superfluous_matrices` shouldn't be empty ?

Reply:
`superfluous_matrices` are the matrices that should no longer exit following the new PR deployment. 
So if you don't have an Error 404 the `else` will execute raising an error.

Reply:
Okay I see